src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/EnvironmentPropertyEntityBuilder.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.tasks.zip.builder;
+
+import com.ca.apim.gateway.cagatewayconfig.tasks.zip.beans.Bundle;
+import com.ca.apim.gateway.cagatewayconfig.util.IdGenerator;
+import org.w3c.dom.Document;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class EnvironmentPropertyEntityBuilder implements EntityBuilder {
+    private final Document document;
+    private final IdGenerator idGenerator;
+
+    EnvironmentPropertyEntityBuilder(Document document, IdGenerator idGenerator) {
+        this.document = document;
+        this.idGenerator = idGenerator;
+    }
+
+    public List<Entity> build(Bundle bundle) {
+        return bundle.getEnvironmentProperties().entrySet().stream()
+                .filter(propertyEntry -> propertyEntry.getKey().startsWith("gateway."))
+                .map(propertyEntry ->
+                        buildEnvironmentPropertyEntity(propertyEntry.getKey().substring(8))
+                ).collect(Collectors.toList());
+    }
+
+    private Entity buildEnvironmentPropertyEntity(String name) {
+        String id = idGenerator.generate();
+        Entity entity = new Entity("CLUSTER_PROPERTY", "ENV." + name, id, ClusterPropertyEntityBuilder.buildClusterPropertyElement("ENV." + name, id, "ENV.gateway." + name, document, new HashMap<String, String>() {{
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Use another way to initialize this instance. ([squid:S3599](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS3599))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=50&issues=AWVI9pWslRPS_ea-RdK1&open=AWVI9pWslRPS_ea-RdK1)

[ISSUE_KEY]: AWVI9pWslRPS_ea-RdK1